platform/srf06-cc26xx/sensortag/mpu-9250-sensor.c

```diff
@@ -377,6 +432,121 @@ enable_sensor(uint16_t axes)
   }
 }
 /*---------------------------------------------------------------------------*/
+/**
+ * \brief Enable the MAG
+ */
+static void 
+mag_enable(void)
+{
+  uint8_t val;
+  if(power_test() != 0) {
+    /* Set magnetometer configurtation, read mode and data resolution */
+    if(!set_bypass()) {
+      return;
+    }
+    SENSOR_SELECT_MAG();  
+    val = (scale << 4) | mode; 
+    sensor_common_write_reg(MAG_CNTL1, &val, 1);
+    delay_ms(10);
+    val = 0;
+    sensor_common_read_reg(MAG_WHO_AM_I, &val, 1);
+    
+  }
+  SENSOR_DESELECT();
+}
+/*---------------------------------------------------------------------------*/
+/**
+* \brief  Shift the I2C bus to control the magnetomoter
+* \return true if success
+*/
+static bool 
+set_bypass(void)
+{
+  bool success;
+  uint8_t val;
+
+  SENSOR_SELECT();
+  delay_ms(100);
+  val = EN_BYPASS;
+  sensor_common_write_reg(INT_PIN_CFG, &val, 1);
+  delay_ms(10);
+  val = 0;
+  sensor_common_read_reg(INT_PIN_CFG, &val, 1);
+  SENSOR_DESELECT();
+  if(val == 0x22) {
+    success = true;
+  }
+  else {
+    success = false;
+  }
+  return success;
+}
+/*---------------------------------------------------------------------------*/
+/**
+ * \brief Read data from the magnetometer - X, Y, Z - 3 words
+ * \param data for storing values
+ * \return magnetometer status, 0 if no error
+ */
+uint8_t 
+mag_read(int16_t *data)
+{
+  uint8_t val;
+  uint8_t rawData[7];
+  magStatus = MAG_NO_POWER;
+  if(power_test() != 0)
+    magStatus = MAG_STATUS_OK;
```

Comment:
I think that `magStatus` should be initiated to `MAG_STATUS_OK`.  And, if `power_test()` is zero, then it should be changed to `MAG_NO_POWER`.  That sequence is a better match to how `magStatus` is used in the rest of this function.

Reply:
i agree, thank you